pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go

```diff
@@ -105,6 +111,61 @@ func (p *partitionedStreamClient) Heartbeat(
 	return nil
 }
 
+// postgresURL converts an SQL serving address into a postgres URL. In order to connect securely
+// using postgres, this method will create temporary on-disk copies of
+// certain embedded security certificates. The certificates will be created in a new temporary
+// directory. The returned cleanup function will delete this temporary directory.
+func postgresURL(servingAddr string, user *url.Userinfo) (url.URL, func(), error) {
+	host, port, err := net.SplitHostPort(servingAddr)
+	if err != nil {
+		return url.URL{}, func() {}, err
+	}
+
+	tempDir, err := ioutil.TempDir("", fileutil.EscapeFilename("StreamClientPGConnection"))
+	if err != nil {
+		return url.URL{}, func() {}, err
+	}
+
+	// This CA is the one used by the SQL client driver to authenticate KV nodes on the host cluster.
+	caPath := filepath.Join(security.EmbeddedCertsDir, security.EmbeddedCACert)
+	tempCAPath, err := securitytest.RestrictedCopy(caPath, tempDir, "ca")
+
+	if err != nil {
+		return url.URL{}, func() {}, err
+	}
+	// This CA is the one used by the SQL client driver to authenticate SQL tenant servers.
+	tenantCAPath := filepath.Join(security.EmbeddedCertsDir, security.EmbeddedTenantCACert)
```

Comment:
Do we ever connect to or as a tenant? isn't it always host-to-host connections, even if we're streaming a tenant?